src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Cohost/AbstractCohostDocumentEndpoint.cs

```diff
@@ -22,6 +22,15 @@ internal abstract class AbstractCohostDocumentEndpoint<TRequest, TResponse>(
             return SpecializedTasks.Default<TResponse>();
         }
 
+        if (context.TextDocument.Project.FilePath is null)
+        {
+            // If the project file path is null, we can't compute the hint name, so we can't handle the request.
+            // This is likely a file in the misc files project, which we don't support yet anyway.
+            // TODO: Expose context.TextDocument.Project.Solution.WorkspaceKind through our EA to confirm?
```

Comment:
FWIW, I think it'd be reasonable to add the workspace kind to `SolutionServices` and take it through Roslyn API review.